### ISSUE TITLE
[Bug] Adding a custom platform mapping on a previously-scanned folder fails

### ISSUE THREAD

**Issue by DragRedSim** [zurdi15/romm#620]
**RomM version**
2.3.1 (Docker)

**Describe the bug**
If a folder has already been scanned under one name, and then a mapping is later added to an IGDB slug, the database will throw an error.

**To Reproduce**
1. Set up, and scan a rom directory. In this case, mine is named 'tg-16cd'.
2. In config.yml, add a mapping as follows: `tg-16cd: 'turbografx-16-slash-pc-engine-cd'`
3. Restart the container.
4. Attempt to scan the Tg 16cd platform.

**Expected behavior**
For the folder to scan, and update according to the IGDB mapping.

**Actual behavior**
```
SQLite3 throws an error:
sqlalchemy.exc.IntegrityError: (sqlite3.IntegrityError) UNIQUE constraint failed: platforms.fs_slug

[SQL: INSERT INTO platforms (slug, fs_slug, name, igdb_id, sgdb_id, logo_path, n_roms) VALUES (?, ?, ?, ?, ?, ?, ?)]

[parameters: ('turbografx-16-slash-pc-engine-cd', 'tg-16cd', 'Turbografx-16/PC Engine CD', 150, None, 'default/default/cover/small.png', 0)]
```

**Additional context**
I'm pretty sure this is triggered by the attempt to add a new platform with the new mapping that was configured after the first scan. 
https://github.com/zurdi15/romm/blob/4135ee52637a2c47610977b9d20a5d116b9b3751/backend/endpoints/scan.py#L46-L47
Basically, there's no wrapping in the dbh to check if there is already a database entry for that `fs_slug`. 
https://github.com/zurdi15/romm/blob/4135ee52637a2c47610977b9d20a5d116b9b3751/backend/handler/db_handler.py#L36-L39

An alternate logic flow I would suggest is:
```
def add_platform(self, platform: Platform, session: Session = None):

        try:

                platform_entry = get_platform_by_fs_slug(platform.fs_slug)

        if (platform_entry is not Null)

                delete_platform(platform.fs_slug)

        return session.merge(platform)
```

Though it may be more efficient to update the already-existing platform entry.

**Comment by zurdi15** [comment]
Hi! this was already fixed in #603 ! I just tested your specific case and works as expected :)

Issue will be automatically closed when releasing the new version

**Comment by gantoine** [comment]
Fixes in the latest release!
